tests/bears/BearTest.py

```diff
@@ -268,6 +269,25 @@ def test_new_result(self):
         self.assertEqual(result, expected)
 
 
+class BrokenReadHTTPResponse(BytesIO):
+
+    def __init__(self, chunks, *args, **kwargs):
+        self.read_count = 0
+        self.chunks = chunks
+
+    def read(self, *args, **kwargs):
+        # A HTTPResponse will return an empty string when you read from it
+        # after the socket has been closed.
+        if self.closed:
+            return b''
+
+        if self.read_count == len(self.chunks):
+            raise requests.exceptions.ReadTimeout('Fake read timeout')
+
+        self.read_count += 1
+        return self.chunks[self.read_count - 1]
```

Comment:
It doesnt.  `read_count` starts at 0, but is incremented on the line above here, so we need a `-1` here to index from 0.
Anyway, I've added a second chunk so that is more clear that the writer and reader are seeing the same content in the same order, until the exception happens.